class-onsite-example-gateway.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+use Give\Donations\Models\Donation;
+use Give\Donations\Models\DonationNote;
+use Give\Donations\ValueObjects\DonationStatus;
+use Give\Framework\Exceptions\Primitives\Exception;
+use Give\Framework\PaymentGateways\Commands\GatewayCommand;
+use Give\Framework\PaymentGateways\Commands\PaymentComplete;
+use Give\Framework\PaymentGateways\Commands\PaymentRefunded;
+use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\Framework\PaymentGateways\PaymentGateway;
+
+/**
+ * @inheritDoc
+ */
+class ExampleGatewayOnsiteClass extends PaymentGateway
+{
+    /**
+     * @inheritDoc
+     */
+    public static function id(): string
+    {
+        return 'onsite-example-test-gateway';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getId(): string
+    {
+        return self::id();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getName(): string
+    {
+        return __('Example Gateway - Onsite', 'example-give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPaymentMethodLabel(): string
+    {
+        return __('Example Gateway - Onsite', 'example-give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getLegacyFormFieldMarkup(int $formId, array $args): string
+    {
+        // Step 1: add any gateway fields to the form using html.
+        // Step 2: you can send this data to the $gatewayData param using the filter `givewp_create_payment_gateway_data_{gatewayId}`.
+        return "<div><input type='text' name='example-gateway-id' placeholder='Example gateway field' /></div>";
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function createPayment(Donation $donation, $gatewayData): GatewayCommand
+    {
+        try {
+            // Step 1: Validate any data passed from the gateway fields in $gatewayData.  Throw the PaymentGatewayException if the data is invalid.
+            if (empty($gatewayData['example_payment_id'])) {
+                throw new PaymentGatewayException('Example payment ID is required.');
+            }
+
+            // Step 2: Create a payment with your gateway.
+            $response = $this->exampleRequest(['transaction_id' => $gatewayData['example_payment_id']]);
+
+            // Step 3: Return a command to complete the donation.
+            return new PaymentComplete($response['transaction_id']);
```

Comment:
Should we leave a note here that `PaymentProcessing` can also be used if the payment needs to be marked as complete later via webhooks?

Reply:
yes. let's add that, @JasonTheAdams is that something I can do?

Reply:
@Benunc Sure! Just add a comment with a line identical to this one that uses `PaymentProcessing` instead. Simple as that! And, of course, explain _why_ one would use that command. 😁